File: examples/inline/python/integrations/adspirer/002-use-with-agent.py
from google.adk.agents import Agent
from google.adk.tools.mcp_tool import McpToolset, StreamableHTTPConnectionParams

ADSPIRER_ACCESS_TOKEN = "YOUR_ADSPIRER_ACCESS_TOKEN"

root_agent = Agent(
    model="gemini-flash-latest",
    name="advertising_agent",
    instruction=(
        "You are an advertising agent that helps users create, manage, "
        "and optimize ad campaigns across Google Ads, Meta Ads, "
        "LinkedIn Ads, and TikTok Ads."
    ),
    tools=[
        McpToolset(
            connection_params=StreamableHTTPConnectionParams(
                url="https://mcp.adspirer.com/mcp",
                headers={
                    "Authorization": f"Bearer {ADSPIRER_ACCESS_TOKEN}",
                },
            ),
        )
    ],
)

File: examples/inline/typescript/integrations/google-developer-knowledge/002-use-with-agent.ts
import { LlmAgent, MCPToolset } from "@google/adk";

const DEVELOPER_KNOWLEDGE_API_KEY = "YOUR_DEVELOPER_KNOWLEDGE_API_KEY";

const rootAgent = new LlmAgent({
    model: "gemini-flash-latest",
    name: "google_knowledge_agent",
    instruction: "Search Google developer documentation for implementation guidance.",
    tools: [
        new MCPToolset({
            type: "StreamableHTTPConnectionParams",
            url: "https://developerknowledge.googleapis.com/mcp",
            transportOptions: {
                requestInit: {
                    headers: {
                        "X-Goog-Api-Key": DEVELOPER_KNOWLEDGE_API_KEY,
                    },
                },
            },
        }),
    ],
});

export { rootAgent };